src/libsystemd/sd-event/sd-event.c

```diff
@@ -4237,6 +4222,21 @@ static int source_dispatch(sd_event_source *s) {
 
         s->dispatching = false;
 
+        if (saved_type != SOURCE_POST) {
+                sd_event_source *z;
+
+                /* If we execute a non-post source, let's mark all post sources as pending. */
+
+                SET_FOREACH(z, saved_event->post_sources) {
+                        if (event_source_is_offline(z))
+                                continue;
+
+                        r = source_set_pending(z, true);
+                        if (r < 0)
+                                return r;
+                }
+        }
```

Comment:
not sure how i feel about this one. this does feel like a compat issue to some point (as sd_event_source_get_pending() would return false now, for all post soruces in any event handler, where it previously said true)

i think it'd feel more comfortable if we'd keep this as above, but maybe just create post event sources as pending, unless the event loop has not been started yet.